go/test/endtoend/cluster/etcd_process.go

```diff
@@ -0,0 +1,160 @@
+limitations under the License.
+
+Runs etcd process and makes entry for global and zone
+*/
+
+package cluster
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+	"os/exec"
+	"path"
+	"strings"
+	"syscall"
+	"time"
+
+	"vitess.io/vitess/go/vt/log"
+)
+
+// EtcdProcess is a generic handle for a running Etcd .
+// It can be spawned manually
+type EtcdProcess struct {
+	Name               string
+	Binary             string
+	DataDirectory      string
+	ListenClientURL    string
+	AdvertiseClientURL string
+	Port               int
+	Host               string
+	VerifyURL          string
+
+	proc *exec.Cmd
+	exit chan error
+}
+
+// Setup spawns a new etcd service and initializes it with the defaults.
+// The service is kept running in the background until TearDown() is called.
+func (etcd *EtcdProcess) Setup() (err error) {
+	etcd.proc = exec.Command(
+		etcd.Binary,
+		"--data-dir", etcd.DataDirectory,
+		"--listen-client-urls", etcd.ListenClientURL,
+		"--advertise-client-urls", etcd.AdvertiseClientURL,
+	)
+
+	etcd.proc.Stderr = os.Stderr
+	etcd.proc.Stdout = os.Stdout
+
+	etcd.proc.Env = append(etcd.proc.Env, os.Environ()...)
+
+	log.Infof("%v %v", strings.Join(etcd.proc.Args, " "))
+	println("Starting etcd with args " + strings.Join(etcd.proc.Args, " "))
+	err = etcd.proc.Start()
+	if err != nil {
+		return
+	}
+
+	etcd.exit = make(chan error)
+	go func() {
+		etcd.exit <- etcd.proc.Wait()
+	}()
+
+	timeout := time.Now().Add(60 * time.Second)
+	for time.Now().Before(timeout) {
+		if etcd.IsHealthy() {
+			return
+		}
+		select {
+		case err := <-etcd.exit:
+			return fmt.Errorf("process '%s' exited prematurely (err: %s)", etcd.Name, err)
+		default:
+			time.Sleep(300 * time.Millisecond)
+		}
+	}
+
+	return fmt.Errorf("process '%s' timed out after 60s (err: %s)", etcd.Name, <-etcd.exit)
+}
+
+// TearDown shutdowns the running mysqld service
+func (etcd *EtcdProcess) TearDown(Cell string) error {
+	if etcd.proc == nil || etcd.exit == nil {
+		return nil
+	}
+
+	etcd.removeTopoDirectories(Cell)
+
+	// Attempt graceful shutdown with SIGTERM first
+	etcd.proc.Process.Signal(syscall.SIGTERM)
+	os.RemoveAll(path.Join(os.Getenv("VTDATAROOT"), "etcd"))
+	select {
+	case err := <-etcd.exit:
+		etcd.proc = nil
+		return err
+
+	case <-time.After(10 * time.Second):
+		etcd.proc.Process.Kill()
+		etcd.proc = nil
+		return <-etcd.exit
+	}
+
+}
+
+// IsHealthy function checks if etcd server is up and running
+func (etcd *EtcdProcess) IsHealthy() bool {
+	resp, err := http.Get(etcd.VerifyURL)
+	if err != nil {
+		return false
+	}
+	if resp.StatusCode == 200 {
+		return true
+	}
+	return false
+}
+
+func (etcd *EtcdProcess) removeTopoDirectories(Cell string) {
+	etcd.ManageTopoDir("rmdir", "/vitess/global")
+	etcd.ManageTopoDir("rmdir", "/vitess/"+Cell)
+}
+
+// ManageTopoDir creates global and zone in etcd2
+func (etcd *EtcdProcess) ManageTopoDir(command string, directory string) error {
+	tmpProcess := exec.Command(
```

Comment:
you can save time by naming such variables with single letters - like `p` for process. That is acceptable.